src/main/java/seedu/address/model/person/Name.java

```diff
@@ -38,7 +38,23 @@ public static boolean isValidName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
-
+    /**
+     * Compares this Name object with the specified Name object for order based on their full names,
+     * ignoring case considerations.
+     *
+     * <p>Returns a negative integer, zero, or a positive integer as this Name object is less than, equal to,
+     * or greater than the specified Name object.
+     *
+     * <p>The comparison is based on lexicographic ordering of the full names of the Names.
+     * The comparison is case-insensitive.
+     *
+     * @param other The Name object to be compared.
+     * @return A negative integer if this Name is less than the specified Name, zero if they are equal, or a
+     *     positive integer if this Name is greater than the specified Name.
+     */
+    public int compare(Name other) {
```

Comment:
Which part of the code are we using the name compare for? Just wondering

Reply:
Edit: I see that it's for ensuring that the person list stays sorted, correct me if I'm wrong!

Reply:
Yup its to ensure that the person list stays sorted. For now, the sorting is only done at insertion. There is no additional check to re-sort the order somehow gets mixed up. But I can't think of any scenario that could cause this, other than if someone tampers with the data file directly.

Reply:
Yup agreed, it would remain sorted, looks good!